### PR TITLE
Add a non-blocking binding to 'readdir'

### PR DESCRIPTION
**Pull request by yallop**

### CODE REVIEW

**Comment by dsheets** [comment]
Thanks!

File: Makefile [hunk 1]
```diff
@@ -7,12 +7,17 @@ OCAML_LIB_DIR=$(shell ocamlc -where)
...
+
+OCAMLBUILD=CTYPES_LIB_DIR=$(CTYPES_LIB_DIR) \
+           LWT_LIB_DIR=$(LWT_LIB_DIR)       \
+           OCAML_LIB_DIR=$(OCAML_LIB_DIR)   \
+             ocamlbuild -use-ocamlfind -classic-display
 
 WITH_UNIX=$(shell ocamlfind query \
             ctypes unix unix-type-representations unix-errno.unix \
             > /dev/null 2>&1 ; echo $$?)
+WITH_LWT=$(shell ocamlfind query lwt unix > /dev/null 2>&1 ; echo $$?)
```
Comment (samoht) [inline Makefile]: I'm not very fond of this build system magic. This make the life harder to opam users to know exactly which depots needs to be installed to enable the given feature.

(this is just a side-node, I have unfortunately no better proposal, although passing env variables to `make` could also be a solution)

Comment (yallop) [inline Makefile]: The `lwt` depopt isn't too bad: the `unix-dirent.lwt` subpackage will be built iff `lwt` is installed.  For the `unix-dirent.unix` subpackage the relationship between depopts and build outputs is a bit less obvious.

One possibility is to have a virtual OPAM package `ocaml-unix-dirent-unix` or somesuch, which depends non-optionally on `ctypes`, `unix-errno` and `unix-type-representations`, and make *that* package the depopt of this package (`ocaml-unix-dirent`).

Comment (dsheets) [inline Makefile]: I agree, it's horrible (and I've just made it more magical in a recent commit). I don't really want to create more opam packages to represent these features... need feature support in opam but I don't know the status of ocaml/opam#1985.

**Pull request merged**